flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/functions/casting/StructuredToStringCastRule.java

```diff
@@ -0,0 +1,233 @@
+        f2Value$293 = ((org.apache.flink.table.data.binary.BinaryStringData) _myInput.getString(2));
+        builder$287.append("c=" + f2Value$293);
+    } else {
+        builder$287.append("c=" + "NULL");
+    }
+    builder$287.append(", ");
+    org.apache.flink.table.data.ArrayData f3Value$295 = null;
+    boolean f3IsNull$296 = _myInput.isNullAt(3);
+    if (!f3IsNull$296) {
+        f3Value$295 = _myInput.getArray(3);
+        isNull$6 = f3IsNull$296;
+        if (!isNull$6) {
+            builder$297.setLength(0);
+            builder$297.append("[");
+            for (int i$299 = 0; i$299 < f3Value$295.size(); i$299++) {
+                if (i$299 != 0) {
+                    builder$297.append(", ");
+                }
+                org.apache.flink.table.data.binary.BinaryStringData element$300 = org.apache.flink.table.data.binary.BinaryStringData.EMPTY_UTF8;
+                boolean elementIsNull$301 = f3Value$295.isNullAt(i$299);
+                if (!elementIsNull$301) {
+                    element$300 = ((org.apache.flink.table.data.binary.BinaryStringData) f3Value$295.getString(i$299));
+                    builder$297.append(element$300);
+                } else {
+                    builder$297.append("NULL");
+                }
+            }
+            builder$297.append("]");
+            java.lang.String resultString$298;
+            resultString$298 = builder$297.toString();
+            result$7 = org.apache.flink.table.data.binary.BinaryStringData.fromString(resultString$298);
+            isNull$6 = result$7 == null;
+        } else {
+            result$7 = org.apache.flink.table.data.binary.BinaryStringData.EMPTY_UTF8;
+        }
+        builder$287.append("d=" + result$7);
+    } else {
+        builder$287.append("d=" + "NULL");
+    }
+    builder$287.append("}");
+    java.lang.String resultString$288;
+    resultString$288 = builder$287.toString();
+    result$1 = org.apache.flink.table.data.binary.BinaryStringData.fromString(resultString$288);
+
+     */
+    @Override
+    protected String generateCodeBlockInternal(
+            CodeGeneratorCastRule.Context context,
+            String inputTerm,
+            String returnVariable,
+            LogicalType inputLogicalType,
+            LogicalType targetLogicalType) {
+        StructuredType inputStructuredType = (StructuredType) inputLogicalType;
+
+        final String builderTerm = newName("builder");
+        context.declareClassField(
+                className(StringBuilder.class), builderTerm, constructorCall(StringBuilder.class));
+
+        final String resultStringTerm = newName("resultString");
+        final int length = LogicalTypeChecks.getLength(targetLogicalType);
+        final CastRuleUtils.CodeWriter writer =
+                new CastRuleUtils.CodeWriter()
+                        .stmt(methodCall(builderTerm, "setLength", 0))
+                        .stmt(methodCall(builderTerm, "append", strLiteral("{")));
```

Comment:
sorry, for the last minute suggestion. but how about we use `(...)` instead of `{...}` to distinguish structured types from maps. then they are also closer to rows which also use `(...)`